extmod/vfs_posix_file.c

```diff
@@ -188,7 +188,7 @@ STATIC mp_uint_t vfs_posix_file_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_
             return 0;
         case MP_STREAM_GET_FILENO:
             return o->fd;
-        #if MICROPY_PY_SELECT
+        #if MICROPY_PY_SELECT && !MICROPY_PY_SELECT_POSIX_OPTIMISATIONS
```

Comment:
This looks unrelated.

Reply:
Yes, I'm keeping track of the file as other updates have been made, this change was made in this commit:
https://github.com/micropython/micropython/commit/ef71028f77e4a6fc449c64906ebad0a160f35240